### PR TITLE
Add scheduler unit test for propagation of dispatch information to jobs file

### LINKED ISSUE #6796: Test propagation of dispatch information to jobs file

### PR DESCRIPTION
**Pull request by jonathan-eq**
**Issue**
Resolves #6796

### CODE REVIEW

File: tests/unit_tests/scheduler/test_scheduler.py [hunk 1]
```diff
@@ -33,39 +33,44 @@ def create_jobs_json(path: Path, steps: Sequence[ForwardModel]) -> None:
                 "executable": step.executable,
                 "argList": step.arglist,
             }
-            for step in steps
+            for step in realization.forward_models
```
Comment (berland) [inline tests/unit_tests/scheduler/test_scheduler.py:36]: `for forward_model in realization.forward_models`

File: tests/unit_tests/scheduler/test_scheduler.py [hunk 2]
```diff
@@ -33,39 +33,44 @@ def create_jobs_json(path: Path, steps: Sequence[ForwardModel]) -> None:
...
         "run_id": "0",
         "ert_pid": "0",
-        "real_id": "0",
+        "real_id": str(realization.iens),
     }
-
-    with open(path / "jobs.json", "w") as f:
+    realization_run_path = Path(realization.run_arg.runpath)
+    realization_run_path.mkdir()
+    with open(realization_run_path / "jobs.json", "w") as f:
```
Comment (berland) [inline tests/unit_tests/scheduler/test_scheduler.py:44]: `encoding="utf-8"`

File: tests/unit_tests/scheduler/test_scheduler.py [hunk 3]
```diff
@@ -33,39 +33,44 @@ def create_jobs_json(path: Path, steps: Sequence[ForwardModel]) -> None:
...
         json.dump(jobs, f)
 
 
 @pytest.fixture
 def realization(storage, tmp_path):
     ensemble = storage.create_experiment().create_ensemble(name="foo", ensemble_size=1)
+    return create_stub_realization(ensemble, tmp_path, 0)
+
 
+def create_stub_realization(ensemble, base_path: Path, iens):
```
Comment (berland) [inline tests/unit_tests/scheduler/test_scheduler.py:54]: `-> Realization`

File: tests/unit_tests/scheduler/test_scheduler.py [hunk 4]
```diff
@@ -106,5 +113,41 @@ async def test_cancel(tmp_path: Path, realization):
...
+    realizations = [
+        create_stub_realization(ensemble, tmp_path, iens)
+        for iens in range(ensemble_size)
+    ]
+
+    sch = scheduler.Scheduler()
+    sch.set_ee_info(test_ee_uri, test_ens_id, test_ee_cert, test_ee_token)
+
+    for realization in realizations:
+        sch.add_realization(realization, callback_timeout=lambda _: None)
```
Comment (berland) [inline tests/unit_tests/scheduler/test_scheduler.py:140]: I think the callback_timeout defaults to None so you don't have to specify it.

Comment (jonathan-eq) [reply tests/unit_tests/scheduler/test_scheduler.py:140]: I tried, and it does not default to None. It is not in use yet, so should I make it default to None or pass in the lambda None?

Comment (berland) [reply tests/unit_tests/scheduler/test_scheduler.py]: ah, there is a missing `= None` in the scheduler.py.

**Comment by jonathan-eq** [comment]
It is always macos py3.10 causing problems..

File: tests/unit_tests/scheduler/test_scheduler.py [hunk 5]
```diff
@@ -21,50 +20,55 @@ def create_bash_step(script: str) -> ForwardModel:
...
         job_name="",
     )
 
-    return Realization(
-        iens=0,
+    realization = Realization(
+        iens=iens,
         forward_models=[],
         active=True,
         max_runtime=None,
```
Comment (xjules) [inline tests/unit_tests/scheduler/test_scheduler.py:66]: If we want to keep it "general" shall max_runtime be a parameter too?

**Review by xjules — COMMENTED** [review]
Looks good @jonathan-eq ! Just had a small comment.

File: tests/unit_tests/scheduler/test_scheduler.py (hunk 5 above)
Comment (jonathan-eq) [reply tests/unit_tests/scheduler/test_scheduler.py:57]: There are currently no tests that use max_runtime, so I think it is better to add that later.

Comment (jonathan-eq) [reply tests/unit_tests/scheduler/test_scheduler.py:57]: We came to an agreement to skip this

**Pull request merged**